packages/firestore/src/remote/remote_store.ts

```diff
@@ -374,127 +437,28 @@ export class RemoteStore {
    * RemoteEvent, and passes that on to the listener, which is typically the
    * SyncEngine.
    */
-  private handleWatchChangeBatch(
-    snapshotVersion: SnapshotVersion,
-    changes: WatchChange[]
-  ): Promise<void> {
-    const aggregator = new WatchChangeAggregator(
-      snapshotVersion,
-      this.listenTargets,
-      this.pendingTargetResponses
+  private raiseWatchSnapshot(snapshotVersion: SnapshotVersion): Promise<void> {
+    const remoteEvent = this.watchChangeAggregator.createRemoteEvent(
```

Comment:
Might be worth an assert that snapshotVersion != SnapshotVersion.MIN (which would fail with the current existence filter code, I believe).